requests/adapters.py

```diff
@@ -436,18 +437,21 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
 
         try:
             if not chunked:
-                resp = conn.urlopen(
-                    method=request.method,
-                    url=url,
-                    body=request.body,
-                    headers=request.headers,
-                    redirect=False,
-                    assert_same_host=False,
-                    preload_content=False,
-                    decode_content=False,
-                    retries=self.max_retries,
-                    timeout=timeout
-                )
+                try:
+                    resp = conn.urlopen(
+                        method=request.method,
+                        url=url,
+                        body=request.body,
+                        headers=request.headers,
+                        redirect=False,
+                        assert_same_host=False,
+                        preload_content=False,
+                        decode_content=False,
+                        retries=self.max_retries,
+                        timeout=timeout
+                    )
+                except LocationParseError as e:
+                    raise InvalidURL(e, request=request)
```

Comment:
Since `InvalidURL` is a `socket.error`, it is caught an re-raised as a `ConnectionError` below. 
This means that currently, a user will see both an `InvalidURL` and a `ConnectionError` in the stack trace. 
Alternatively, the `InvalidURL` could be caught and re-raised as-is; I'm not sure of the best way to handle this.